okhttp/src/main/kotlin/okhttp3/internal/connection/CallConnectionUser.kt

```diff
@@ -0,0 +1,140 @@
+package okhttp3.internal.connection
+
+import java.io.IOException
+import java.net.InetAddress
+import java.net.Proxy
+import java.net.Socket
+import okhttp3.Connection
+import okhttp3.ConnectionListener
+import okhttp3.EventListener
+import okhttp3.Handshake
+import okhttp3.HttpUrl
+import okhttp3.Protocol
+import okhttp3.Route
+import okhttp3.internal.http.RealInterceptorChain
+
+/**
+ * A connection user that is a specific [RealCall].
+ */
+internal class CallConnectionUser(
+  private val call: RealCall,
+  private val poolConnectionListener: ConnectionListener,
```

Comment:
Should the concrete instances on this interface be responsible for notifying the connection pool?

Seems like logic should match regardless of the impl.

Maybe pull out or wrap with a poolconnectionuser?

#tioli